internal/cmd/audit_trail/list.go

```diff
@@ -0,0 +1,248 @@
+package audittrail
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"slices"
+
+	"github.com/pkg/errors"
+	"github.com/shurcooL/graphql"
+	"github.com/spacelift-io/spacectl/client/structs"
+	"github.com/spacelift-io/spacectl/internal"
+	"github.com/spacelift-io/spacectl/internal/cmd"
+	"github.com/spacelift-io/spacectl/internal/cmd/authenticated"
+	"github.com/urfave/cli/v2"
+)
+
+var defaultOrder = structs.QueryOrder{
+	Field:     "createdAt",
+	Direction: "DESC",
+}
+
+func listAuditTrails() cli.ActionFunc {
+	return func(cliCtx *cli.Context) error {
+		outputFormat, err := cmd.GetOutputFormat(cliCtx)
+		if err != nil {
+			return err
+		}
+
+		var limit *uint
+		if cliCtx.IsSet(cmd.FlagLimit.Name) {
+			if cliCtx.Uint(cmd.FlagLimit.Name) == 0 {
```

Comment:
Just a thought, do you think these are useful? I think we have same checks in other list functions, but I'm just curious whats your opinion on checks like this.

Reply:
I think most of those checks are ok. I like showing reasonable error message when input has no sense. However I'm not a fan of this check:
```go
if cliCtx.Uint(cmd.FlagLimit.Name) == 0 {
	return fmt.Errorf("limit must be greater than 0")
}
```
and i added it only because we have it in other places. In my opinion `0` should be allowed value. We event say that `0` is default 
```
   --limit value               [Optional] Limit the number of items to return (default: 0)
```
so not allowing explicit `0` is a little bit weird to me

Reply:
I'm cool with removing such checks